js/components/matrix.js

```diff
@@ -0,0 +1,366 @@
+
+        this.value[row.id || row.label] = col.id || col.label;
+        this.clickHandler(row, col, rowIndex, colIndex);
+      }
+    },
+
+    /**
+     * Set default undefined value for each row if value is blank
+     *
+     * @returns {undefined}
+     */
+    setDefaultValue: function() {
+      var $vm = this;
+
+      $vm.value = {};
+
+      _.forEach(this.rowOptions, function(row) {
+        $vm.$set($vm.value, row.id || row.label, undefined);
+      });
+    },
+
+    /**
+     * Set value for each row as per selection and default value
+     *
+     * @returns {undefined}
+     */
+    setValue: function() {
+      var $vm = this;
+
+      if ($vm.value === undefined || _.isEmpty($vm.value)) {
+        this.setDefaultValue();
+      } else {
+        if (typeof $vm.value === 'string') {
+          $vm.value = JSON.parse($vm.value);
+        }
+
+        _.forOwn($vm.value, function(key, value) {
+          var row = _.find($vm.rowOptions, function(row) {
+            return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === value : row.label === value;
+          });
+
+          var rowIndex = _.findIndex($vm.rowOptions, function(row) {
+            return (_.has(row, 'label') && _.has(row, 'id')) ? row.id === value || row.label === value : row.label === value;
+          });
+
+          var col = _.find($vm.columnOptions, function(col) {
+            return (_.has(col, 'label') && _.has(col, 'id')) ? col.id === key : col.label === key;
+          });
+
+          var colIndex = _.findIndex($vm.columnOptions, function(col) {
+            return (_.has(col, 'label') && _.has(col, 'id')) ? col.id === key || col.label === key : col.label === key;
+          });
+
+          // setTimeout using to load all HTML and then execute below piece of code.
+          setTimeout(function() {
+            if (row && col) {
+              $vm.clickHandler(row, col, rowIndex, colIndex);
+            } else if (rowIndex >= 0) {
+              $vm.clickHandler($vm.rowOptions[rowIndex], $vm.columnOptions[colIndex], rowIndex, colIndex);
+            }
+          });
+        });
+      }
+    },
+
+    /**
+     * Clears the value for each row
+     *
+     * @returns {undefined}
+     */
+    clearValue() {
+      if (!this.$refs.matrix) {
+        return;
+      }
+
+      var $vm = this;
+
+      _.forEach(this.rowOptions, function(row, rowIndex) {
+        _.forEach($vm.columnOptions, function(col, colIndex) {
+          var el = $vm.getOptionId(rowIndex, colIndex, 'input');
+
+          $('#' + el).prop('checked', false);
+        });
+      });
+
+      if (this.value === '') {
+        this.setDefaultValue();
+      }
+    },
+
+    /**
+     * Check if all rows has column values or not
+     *
+     * @param {object} val - value of selected options
+     *
+     * @returns {undefined}
+     */
+    checkValue: function(val) {
+      if (typeof val === 'string' && val !== '') {
+        val = JSON.parse(this.value);
```

Comment:
did you unit test with bigger values in cells? what is max limit for txt?

Reply:
@yogeshmsharma-xoriant  We have value here in object format with row and column options.